src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -1116,42 +962,42 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
         SymbolFuncInfo *s = new SymbolFuncInfo;
 
         /********************* New Type Declaration *********************/
-        wasm::emit_b8(m_type_section, m_al, 0x60);
+        m_wa.emit_b8(m_wa.m_type_section, m_al, 0x60);
 
         /********************* Parameter Types List *********************/
         s->referenced_vars.reserve(m_al, x.n_args);
         uint32_t len_idx_type_section_param_types_list =
-                wasm::emit_len_placeholder(m_type_section, m_al);
+                m_wa.emit_len_placeholder(m_wa.m_type_section, m_al);
         for (size_t i = 0; i < x.n_args; i++) {
             ASR::Variable_t *arg = ASRUtils::EXPR2VAR(x.m_args[i]);
             LCOMPILERS_ASSERT(ASRUtils::is_arg_dummy(arg->m_intent));
             m_var_idx_map[get_hash((ASR::asr_t *)arg)] = s->no_of_variables;
-            emit_var_type(m_type_section, arg, s->no_of_variables, false);
+            emit_var_type(m_wa.m_type_section, arg, s->no_of_variables, false);
             if (isRefVar(arg)) {
                 s->referenced_vars.push_back(m_al, arg);
             }
         }
-        wasm::fixup_len(m_type_section, m_al,
+        m_wa.fixup_len(m_wa.m_type_section, m_al,
                             len_idx_type_section_param_types_list);
```

Comment:
The variables noted in https://github.com/lcompilers/lpython/pull/1657#discussion_r1155150521 are needed at several places in the `ASRToWASMVisitor` Class. Thus, I am unable to make them private.

It seems that without those variables being private, we may/might not have truly decoupled `WASMAssembler` from `ASRToWASMVisitor` Class.

Reply:
This is fixed in https://github.com/lcompilers/lpython/pull/1675.